### ISSUE TITLE
Error when cascade deleting record with composite keys

### ISSUE THREAD

**Issue by skolodyazhnyy** [go-rel/rel#190]
It seems delete query is not properly constructed when cascade deleting one-to-many associations.

The query I get looks like this:
```
DELETE FROM `payment_transaction` WHERE (`payment_id`=? AND );
```

Here is code which produces invalid query:

```go
type Payment struct {
	ID           string `db:"id"`
	Transactions []Transaction   `rel:"id" fk:"payment_id"`
}

type Transaction struct {
	RequestID string `db:"request_id,primary"`
	PaymentID string `db:"payment_id,primary"`
	Name      string `db:"name"`
}

payment := &Payment{
	ID: "some-uuid",
	Transactions: []Transaction{},
}

err := repo.Delete(ctx, payment, true)
```

I believe the problem is here: https://github.com/go-rel/rel/blob/master/repository.go#L879, in case associations slice is empty `filterCollection` returns an empty `rel.Eq`.

**Comment by skolodyazhnyy** [comment]
As a quick workaround I've added `repo.Delete(ctx, payment, len(payment.Transactions) != 0)` :)

**Comment by skolodyazhnyy** [comment]
Ah well, after updating to the latest rel it seems like composite keys are not supported in associations. Maybe worth updating https://go-rel.github.io/basics/#primary-key or https://go-rel.github.io/association/.

**Comment by skolodyazhnyy** [comment]
The code above does not work without composite keys either:

```go
type Payment struct {
	ID           string `db:"id"`
	Transactions []Transaction   `rel:"id" fk:"payment_id"`
}

type Transaction struct {
	ID           string `db:"id,primary"`
	RequestID string `db:"request_id"`
	PaymentID string `db:"payment_id"`
	Name      string `db:"name"`
}

payment := &Payment{
	ID: "some-uuid",
	Transactions: []Transaction{},
}

err := repo.Delete(ctx, payment, true)
```

In this case the query is valid, but one of the arguments is missing: `sql: expected 2 arguments, got 1`

**Comment by Fs02** [comment]
> I believe the problem is here: https://github.com/go-rel/rel/blob/master/repository.go#L879, in case associations slice is empty filterCollection returns an empty rel.Eq.

yes, looks like this is the problem

> Ah well, after updating to the latest rel it seems like composite keys are not supported in associations. Maybe worth updating https://go-rel.github.io/basics/#primary-key or https://go-rel.github.io/association/.

What's not supported is assoc of two tables with composite keys, your first example should be working

**Comment by skolodyazhnyy** [comment]
I see you already added fix in #191, it's great, thanks!

I'm not sure about composite keys though. If you try to insert/update example in the PR description you will get panic, which says composite keys are not supported. If you believe scenario in the description should be supported I can open another issue or maybe even PR. Let me know.

**Comment by Fs02** [comment]
Ah, looks like because that method is using a singular `PrimaryField()` or `PrimaryValue()`

Feel free to investigate, PR will be very appreciated 🎉
